ethereumj-core/src/main/java/org/ethereum/sharding/crypto/BLS381.java

```diff
@@ -0,0 +1,53 @@
+package org.ethereum.sharding.crypto;
+
+import java.math.BigInteger;
+
+/**
+ * Common interface for BLS381-12 implementation
+ */
+public interface BLS381 {
+
+    BI generatePrivate();
+
+    BI restorePrivate(BigInteger value);
+
+    BI restorePrivate(byte[] value);
+
+    ECP1Point restoreECP1(byte[] value);
+
+    ECP2Point restoreECP2(byte[] value);
+
+    ECP2Point generator2();
+
+    ECP1Point mapToECP1(byte[] value);
+
+    FP12Point pair(ECP2Point pointECP2, ECP1Point pointECP1);
+
+    interface BI {
+        BigInteger asBigInteger();
+
+        byte[] asByteArray();
+    }
+
+    interface ECP1Point {
+        ECP1Point mul(BI value);
+
+        void add(ECP1Point value);
+
+        BigInteger asBigInteger();
+
+        byte[] asByteArray();
+    }
+
+    interface ECP2Point {
+        ECP2Point mul(BI value);
+
+        void add(ECP2Point value);
+
+        BigInteger asBigInteger();
+
+        byte[] asByteArray();
+    }
+
+    interface FP12Point {}
```

Comment:
This is not a point, actually. Let's call it `FP12`. And I would, also, explicitly add `equals` method to that interface to point it out as an expecting one from underlying implementations